openid-data-classes/src/commonMain/kotlin/at/asitplus/iso/SessionTranscript.kt

```diff
@@ -30,15 +30,19 @@ data class SessionTranscript private constructor(
     val deviceEngagementBytesOid: Int? = 42,
     /** Set to `null` for OID4VP with ISO/IEC 18013-7 */
     val eReaderKeyBytesOid: Int? = 42,
-    /** Set either this or [nfcHandover] or deviceEngagementBytesOid to null for QR engagement */
+    @Deprecated("Use [openId4VpHandover] instead")
     val oid4VPHandover: OID4VPHandover? = null,
-    /** Set either this or [oid4VPHandover] or deviceEngagementBytesOid to null for QR engagement */
+    /** Set either this or [openId4VpHandover] or [deviceEngagementBytesOid] to `null` for QR engagement */
     val nfcHandover: NFCHandover? = null,
     val dcapiHandover: DCAPIHandover? = null,
+    val openId4VpHandover: OpenId4VpHandover? = null,
 ) {
     init {
-        val nrOfHandovers = listOf(oid4VPHandover, nfcHandover, dcapiHandover).count { it != null }
-        check(nrOfHandovers == 1 || (deviceEngagementBytesOid == null && nrOfHandovers == 0)) { "Exactly one handover element must be set (or null for QR Handover)" }
+        val nrOfHandovers = listOf(oid4VPHandover, nfcHandover, dcapiHandover, openId4VpHandover)
+            .count { it != null }
+        check(nrOfHandovers == 1 || (deviceEngagementBytesOid == null && nrOfHandovers == 0)) {
```

Comment:
I really don't like this. we have four nullable constructor parameters that have non-trivial interdependencies. Hide this constructor and create overloaded ones that disallow messing up. Keep one vor backwards compatibility but mark it as deprecated